botstrap.py

```diff
@@ -137,117 +193,289 @@ def get_all_channels_and_categories(self) -> tuple[dict[str, str], dict[str, str
         channels = SilencedDict(name="Channels dictionary")
         categories = SilencedDict(name="Categories dictionary")
 
-        response = self.get(f"guilds/{self.guild_id}/channels")
-        server_channels = response.json()
-
-        for channel in server_channels:
+        for channel in self.get_guild_channels():
             channel_type = channel["type"]
             name = "_".join(part.lower() for part in channel["name"].split(" ")).replace("-", "_")
             if re.match(off_topic_channel_name_regex, name):
                 name = f"off_topic_{off_topic_count}"
                 off_topic_count += 1
 
-            if channel_type == 4:
+            if channel_type == GUILD_CATEGORY_TYPE:
                 categories[name] = channel["id"]
             else:
                 channels[name] = channel["id"]
 
         return channels, categories
 
-    def webhook_exists(self, webhook_id_: int) -> bool:
-        """A predicate that indicates whether a webhook exists already or not."""
-        try:
-            self.get(f"webhooks/{webhook_id_}")
-            return True
-        except HTTPStatusError:
-            return False
+    def get_all_guild_webhooks(self) -> list[dict[str, Any]]:
+        """Lists all the webhooks for the guild."""
+        response = self.get(f"/guilds/{self.guild_id}/webhooks")
+        return response.json()
 
-    def create_webhook(self, name: str, channel_id_: int) -> str:
+    def create_webhook(self, name: str, channel_id: int | str) -> str:
         """Creates a new webhook for a particular channel."""
         payload = {"name": name}
-
-        response = self.post(f"channels/{channel_id_}/webhooks", json=payload)
+        response = self.post(
+            f"/channels/{channel_id}/webhooks",
+            json=payload,
+            headers={"X-Audit-Log-Reason": "Creating webhook as part of PyDis botstrap"},
+        )
         new_webhook = response.json()
+        log.info("Creating webhook: %s has been successfully created.", name)
         return new_webhook["id"]
 
+    def list_emojis(self) -> list[dict[str, Any]]:
+        """Lists all the emojis for the guild."""
+        response = self.get(f"/guilds/{self.guild_id}/emojis")
+        return response.json()
+
+    def get_emoji_contents(self, id_: str | int) -> bytes | None:
+        """Fetches the image data for an emoji by ID."""
+        # Emojis are located at https://cdn.discordapp.com/emojis/{emoji_id}.{ext}
+        response = self.get(f"{self.CDN_BASE_URL}/emojis/{id_!s}.webp")
```

Comment:
Shouldn't fetch webp if you then submit `image/png`, even if Discord is accepting it it's not right.